### ISSUE TITLE
[BUG] generate_addon_csi.sh fails to complete on windows

### ISSUE THREAD

**Issue by mortenn** [harvester/harvester#4848]
**Describe the bug**
generate_addon_csi.sh fails due to an error `integer expression expected` which makes it go into an infinite loop.

**To Reproduce**
1. Use windows 11
2. Follow the docs for setting up [harvester-hci](https://docs.harvesterhci.io/v1.1/rancher/csi-driver#deploy-harvester-csi-driver)
3. Script fails to complete, printing `Getting secret of service account ... on ...` repeatedly

**Expected behavior**
Script works as advertised in docs

**Support bundle**
N/A

**Environment**
 - Harvester ISO version: N/A
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): N/A

**Additional context**
Nothing notable

**Comment by Vicente-Cheng** [comment]
Hi @mortenn, thanks for reporting this issue.

Could you provide what version of harvester-csi-driver you use?
IIRC, we drop Windows support on the recent version.

Also, thanks for the PR, we will check it!

**Comment by mortenn** [comment]
I wasn't using any version of the driver per se, I just downloaded and ran the script in git bash on windows

**Comment by Vicente-Cheng** [comment]
Hi @mortenn, 
Thanks for your reply.

The PR looks good, but could I know more about your use case?
Did you run a Windows VM for the guest cluster and try to use the harvester csi driver to create PVC?

**Comment by mortenn** [comment]
no, I was trying to configure the csi driver for use with a kubernetes cluster deployed from rancher, the windows pc is my desktop computer.

**Comment by Vicente-Cheng** [comment]
> no, I was trying to configure the csi driver for use with a kubernetes cluster deployed from rancher, the windows pc is my desktop computer.

OK, because harvester CSI driver depends on the harvester cluster (means you need to create kubernetes cluster from harvester, you can refer to https://docs.harvesterhci.io/v1.2/rancher/index#deploying-rancher-server) then you can provision pvc on the harvester cluster.

So if you directly provision the Kubernetes cluster on your Windows PC, I am afraid that the harvester CSI driver might not work. 

But your patch seems to make sense; already merged it. Thanks for fixing it!

**Comment by irishgordo** [comment]
The issue **_can be reproduced_** on:
- Windows 11 Pro, [Git For Windows 2.43](https://github.com/git-for-windows/git/releases/tag/v2.43.0.windows.1), [jq 1.7](https://github.com/jqlang/jq/releases/tag/jq-1.7)

The issue **_can not be reproduced_** via [WSL](https://learn.microsoft.com/en-us/windows/wsl/install) on Windows 11 Pro with:
- WSL version 2.0.9.0
- Ubuntu 22.04.3 , w/ Kernel 5.15.133.1-microsoft-standard-WSL2
- Either with Bash:
  - 5.1.16 (bundled with Ubuntu 22.04.3)
  - compiled from tarball, 5.2.21 (to match the same version of Bash included in `Git For Windows 2.43`)

_Reproduction reproduced yielding:_

```bash
micha_427b7f2@WINDOWS-BOX MINGW64 ~/Code
$ ./generate_addon_csi_old_no_fix.sh test default RKE2
NOTE: if your are using RKE2/K3S, please follow this usage: ./generate_addon_csi_old_no_fix.sh <service_account_name> <namespace> rke2/k3s
Creating target directory to hold files in ./tmp/kube...done
Creating a service account in default namespace: test
W0123 13:28:48.215900   15996 helpers.go:704] --dry-run is deprecated and can be replaced with --dry-run=client.
serviceaccount/test created
Creating a rolebinding in default namespace: default-test
W0123 13:28:48.378881    8396 helpers.go:704] --dry-run is deprecated and can be replaced with --dry-run=client.
[rolebinding.rbac.authorization.k8s.io/default-test](http://rolebinding.rbac.authorization.k8s.io/default-test) created
WARNING: version difference between client (1.29) and server (1.25) exceeds the supported minor version skew of +/-1
: integer expression expectedix.sh: line 57: [: 25
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
Getting secret of service account test on default
```

Additional Testing Artifacts (Windows 10 Pro updated to Windows 11 Pro):
![1000009225](https://github.com/harvester/harvester/assets/5370752/cf7099a6-3b53-42a2-ae52-856df27c8a9b)
![1000009224](https://github.com/harvester/harvester/assets/5370752/d4374cb9-9d7e-4192-bf69-34f256d88c1f)
![1000009223](https://github.com/harvester/harvester/assets/5370752/67209a61-a134-413f-bd1a-a1edf1f128cd)
![1000009234](https://github.com/harvester/harvester/assets/5370752/0ec9caa6-6c19-48cd-8512-0f4518b93b96)
![1000009232](https://github.com/harvester/harvester/assets/5370752/e45c872f-1276-4020-84e9-4372de815f44)
![1000009221](https://github.com/harvester/harvester/assets/5370752/101bec1d-e66d-491e-bfb1-7c5074a3dda4)
![1000009233](https://github.com/harvester/harvester/assets/5370752/13a14a39-c02f-4ff7-9c1a-c653b0569d14)
![oldscript-works-for-wsl](https://github.com/harvester/harvester/assets/5370752/7b9534a3-51c5-42a8-ad88-a545bfb3a207)
![windows-stuff-subsystem](https://github.com/harvester/harvester/assets/5370752/f5fbfe91-2a5b-48c4-8cf0-b3e13d2056b9)
![windows-subsystem-linux-installtest-workflow1](https://github.com/harvester/harvester/assets/5370752/1c10ada9-5cb4-4edb-af66-20e3800d27cd)
![windows-subsystem-linux-with-ubuntu-the-old-script-works](https://github.com/harvester/harvester/assets/5370752/fa3e3409-9ed8-4bb8-b69d-132691a0c923)
![wsl-more-1](https://github.com/harvester/harvester/assets/5370752/0101fc8e-0b2b-4430-9600-52b716e80e0a)
![bash-on-wsl-ubuntu-is-5 1 16-wheras-bash-on-git-for-windows-is-5 2 (1)](https://github.com/harvester/harvester/assets/5370752/08759b73-709f-4e69-a40d-bc5eea477c09)
![bash-on-wsl-ubuntu-is-5 1 16-wheras-bash-on-git-for-windows-is-5 2](https://github.com/harvester/harvester/assets/5370752/2603133a-2597-4cd5-894a-760068a01ebf)


This looks good :smile: :+1: !
Thanks again for the contribution @mortenn !
I'll go ahead and close this out
